test/test_macho.rb

```diff
@@ -46,6 +46,16 @@ def test_load_commands
     end
   end
 
+  def test_load_commands_well_ordered
+    filenames = SINGLE_ARCHES.map { |a| fixture(a, "libhello.dylib") }
+
+    filenames.each do |fn|
+      file = MachO::MachOFile.new(fn)
+      offsets = file.load_commands.map(&:view).map(&:offset)
+      assert array_sorted?(offsets)
```

Comment:
I appreciate the effort to create a reusable predicate method for this check. However, I think this test code is not performance critical, thus I'd prefer to use something really simple like:

```ruby
assert_equal offsets.sort, offsets
```

Another advantage is that this by default results in a more useful message if the assertion fails.